articles/tests/test_views.py

```diff
@@ -262,3 +264,307 @@ def test_get_article_with_invalid_slug_fail(self) -> None:
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls.password = fake.password()
+        cls.client = APIClient()
+        cls.user = User.objects.create_user(
+            username=fake.name(), email=fake.email(), password=cls.password
+        )
+        cls.profile = Profile.objects.create(user=cls.user)
+        cls.article = Article.objects.create(
+            title=fake.text(),
+            body=fake.text(),
+            author=cls.user,
+        )
+        cls.comment = Comment.objects.create(
+            author=cls.profile, article=cls.article, body=fake.text()
+        )
+        cls.data = {
+            "body": fake.text(),
+            "article": cls.article,
+            "author": cls.profile,
+        }
+
+    @property
+    def bearer_token(self) -> dict:
+        login_url = reverse("login")
+        response = self.client.post(
+            login_url,
+            data={"email": self.user.email, "password": self.password},
+            format="json",
+        )
+        token = json.loads(response.content).get("access")
+        return {"HTTP_AUTHORIZATION": f"Bearer {token}"}
+
+    def test_create_comment(
+        self,
+    ) -> None:
+        response = self.client.post(
+            reverse("comments", kwargs={"slug": self.article.slug}),
+            data=self.data,
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(str(self.data.get("body")), self.data.get("body"))
+
+    def test_create_comment_without_authentication(
+        self,
+    ) -> None:
+        response = self.client.post(
+            reverse("comments", kwargs={"slug": self.article.slug}),
+            data=self.data,
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertEqual(
+            json.loads(response.content).get("detail"),
+            "Authentication credentials were not provided.",
+        )
+
+    def test_get_all_comments(
+        self,
+    ) -> None:
+        response = self.client.get(
+            reverse("comments", kwargs={"slug": self.article.slug})
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_get_comment(
+        self,
+    ) -> None:
+        response = self.client.get(
+            reverse(
+                "comment-detail",
+                kwargs={
+                    "slug": self.article.slug,
+                    "lookup_id": self.comment.lookup_id,
+                },
+            ),
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_comment_without_authentication(
+        self,
+    ) -> None:
+        response = self.client.get(
+            reverse(
+                "comment-detail",
+                kwargs={
+                    "slug": self.article.slug,
+                    "lookup_id": self.comment.lookup_id,
```

Comment:
Why do you need both the lookup_id and slug for llookup? just one should should suffice

Reply:
Comment must belong to a specific article, hence the slug